Forward fix failures for torch.export switch to predispatch

tugsbayasgalan opened the pull request:
Summary:
Fixes:
- executorch test
- torchrec test

Test Plan: CI

Differential Revision: D57282304

angelayi commented on torch/_export/passes/replace_set_grad_with_hop_pass.py:
```diff
@@ -106,9 +112,7 @@ def _replace_with_hop(node: torch.fx.Node):
-                raise NotImplementedError(
-                    "Cannot replace a call_module with a hop if it has no output. This module will gets DCEed."
-                )
+                node.graph.erase_node(node)
```
idk what this is doing, but will stamp to unblock you from fixing the test failures :P

tugsbayasgalan committed:
Forward fix failures for torch.export switch to predispatch (#126081)

Summary:

Fixes:
- executorch test
- torchrec test

Test Plan: CI

Reviewed By: angelayi, zzh142857

Differential Revision: D57282304
